server/neptune/gateway/event/check_run_handler.go

```diff
@@ -102,27 +121,70 @@ func (h *CheckRunHandler) signalPlanReviewWorkflowChannel(ctx context.Context, e
 	return nil
 }
 
-func (h *CheckRunHandler) signalUnlockWorkflowChannel(ctx context.Context, event CheckRun) error {
-	// Parse out root from check run name
-	matches := checkRunRegex.FindStringSubmatch(event.Name)
-	if len(matches) != 2 {
-		return fmt.Errorf("unable to determine root name")
-	}
-	rootName := matches[checkRunRegex.SubexpIndex("name")]
-	ctx = context.WithValue(ctx, contextInternal.ProjectKey, rootName)
-	err := h.TemporalClient.SignalWorkflow(
+func (h *CheckRunHandler) signalUnlockWorkflowChannel(ctx context.Context, event CheckRun, rootName string) error {
+	err := h.DeploySignaler.SignalWorkflow(
 		ctx,
 		// deploy workflow id is repo||root (the name of the check run is the root)
 		buildDeployWorkflowID(event.Repo.FullName, rootName),
 		// keeping this empty is fine since temporal will find the currently running workflow
 		"",
 		workflows.DeployUnlockSignalName,
 		workflows.DeployUnlockSignalRequest{
-			User: event.User,
+			User: event.User.Username,
 		})
 	if err != nil {
 		return errors.Wrapf(err, "signaling workflow with id: %s", event.ExternalID)
 	}
 	h.Logger.InfoContext(ctx, fmt.Sprintf("Signaled workflow with id %s to unlock", event.ExternalID))
 	return nil
 }
+
+func (h *CheckRunHandler) handleReRequestedRun(ctx context.Context, event CheckRun, rootName string) error {
```

Comment:
nit: Can you group the handle methods together in the file instead of having this one down here?